### PR TITLE
handle different test step types in CallTreeBuilder

### PR DESCRIPTION
**Pull request by jjelschen**
* extracted some code from `TclJvmModelInferrer` for building log entry strings from test steps and other TCL model elements, and put it into the utility class, `TclElementStringifier`, along with other methods for "stringification"
* `CallTreeBuilder` should now handle test steps in both component and macro context, and resolve macro and fixture calls.

Left to do:
* handling configs
* complex test with non-trivial setup, cleanup, and test steps
* test nested macro calls (e.g. macro calls macro calls fixture method)

### CODE REVIEW

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend [hunk 1]
```diff
@@ -645,8 +631,8 @@ class TclJvmModelInferrer extends AbstractModelInferrer {
 			output.trace(step, TEST_STEP_WITH_ASSIGNMENT__VARIABLE, 0) => [
 				// TODO should we use output.declareVariable here?
 				// val variableName = output.declareVariable(step.variableName, step.variableName)
-				val partialCodeLine = '''«operation.returnType.identifier» «step.variable.name» = '''
-				output.appendReporterEnterCall(SemanticUnit.STEP, '''«partialCodeLine.trim» «stepLog.trim»''', variables)
+				val partialCodeLine = step.leftHandSideAssignmentCodeLine
+				output.appendReporterEnterCall(SemanticUnit.STEP, '''«step.leftHandSideAssignmentCodeLine.trim» «step.stringifyTestStep»''', variables)
```
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend]: could use partialCodeLine here, too, right?

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilderTest.xtend [hunk 2]
```diff
@@ -1,166 +1,291 @@
...
+			]
+		]
+
+		// when
+		val actualTree = builderUnderTest.buildCallTree(testCase)
+
+		// then
+		actualTree.children.head.children.head.children => [
+			assertSize(1)
+			head.displayname.assertEquals('com.google.gson.JsonObject myVar = Read jsonObject from <bar>')
```
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilderTest.xtend]: Type of variable in parenthesis at the end? It is more of an implementation detail and should not be presented to the Tester, right?

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend (hunk 1 above)
Comment (gunther-bachmann) [reply tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend]: If the code line differs from the reported string, be careful though ...

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilderTest.xtend (hunk 2 above)
Comment (jjelschen) [reply tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilderTest.xtend]: … it's now `myVar = Read jsonObject from <bar> [com.google.gson.JsonObject]`; this format is also used for `reporterEnterCall(…)` (see above), and there were several test cases covering this, which I adapted, accordingly.

I suppose another option to consider would be to only put the simple type into the log message (i.e. substring starting from lastIndexOf('.'), leaving only `[JsonObject]` as type info suffix), as the type info can get quite long, e.g. for types with type parameters (generics).

**Pull request merged**